cli/src/pcluster/models/cluster.py

```diff
@@ -238,6 +286,33 @@ def status(self):
         """Return the cluster status."""
         return self.stack.status
 
+    @property
+    def bucket(self):
+        """Return a bucket configuration."""
+        # try to get custom bucket name from stack.
+        try:
+            AWSApi.instance().cfn.describe_stack(self.stack_name)
```

Comment:
I don't think we can use self.stack to replace this check, the cases are different for `_get_artifact_dir` and `bucket` property, because `_get_artifact_dir` is executed in pcluster delete/update commands, but `bucket` will be executed in pcluster create/delete/update commands, in pcluster create we don't have stack, in that case we still need to initailize the bucket rather than raising a ClusterActionError.